### ISSUE TITLE
Improve counter-intuitive "save all" for multiple unnamed files

### ISSUE THREAD

**Issue by kriegaex** [notepad-plus-plus/notepad-plus-plus#8283]
Situation: multiple files open, several of them yet unnamed (unsaved).

Action: Select "Save all" or press `Ctrl-Alt-S`.

Expected result:
  * Each previously saved file is saved silently again.
  * For each unsaved file, the file gets the focus (comes to the foreground), so the user can see the content for the file she is going to save and select a new name for next.
  * The user enters the new file name and saves the file.
  * Next unnamed file pops into to the foreground, user can see and save it. And so forth.
  * The title each "save as" pop-up dialogue box should be "save all (files)", not just "save as" like for a single file.

Actual result:
  * Unnamed files do not get focus.
  * Multiple "save file as" dialogues open one after another.
  * The user cannot see what she is saving, only guess or speculate.
  * Only after a file was saved, the name on its tab changes. But still the user does not see the file content.

**Comment by cbckly** [comment]
YES!  I keep hitting Ctrl-Shift-S with the intent to just "Save As" the current document and getting stuck in this confusing loop.    I reassigned it of course, but this default shortcut seems wrongly assigned.

**Comment by sasumner** [comment]
>I reassigned it of course, but this default shortcut seems wrongly assigned.

This comes up often and prior investigation shows that there seems to be no "standard" -- unless you have some evidence of one?

**Comment by kriegaex** [comment]
If it comes up often, would you not say that the users might have a point there? There is not standard across all Windows applications, but I researched a few standard applications and Windows text editors. This is the result for "Save as":
  * Word, Powerpoint, Excel, UltraEdit, TextPad: **F12**
  * Visual Studion Code, Atom: **Ctrl-Shift-S**

In none of those applications Ctrl-Shift-S was "Save all".

**Comment by sasumner** [comment]
The evidence regarding the keycombo is along the lines of what prior evidence shows -- that there is no conclusive standard across Windows for it.  Thus it will remain the way it is in Notepad++.  And of course arguments about keycombos don't get much weight anyway since all keycombos can be reassigned -- if you don't like it, go ahead and reassign it (like you did).

The issue remains open because of the other aspects.

**Comment by kriegaex** [comment]
7 programs which I checked, only two shortcuts, none of them like yours. IMO you are just ignoring evidence in order not to have to change anything. I guess, if I had presented 25 programs and only 24 would have used the same shortcut, your reply would have been the same.

Fine, I have customised the shortcuts already for myself, like I said. I just feel pity for other users who don't want to or know how to customise the product.

**Comment by sasumner** [comment]
>In none of those applications Ctrl-Shift-S was "Save all".

Arguably the most widely used programmer's text editor in the world is Visual Studio's -- which for those that don't know is NOT "Visual Studio Code" -- and it brings to the table:

![image](https://user-images.githubusercontent.com/30118311/118865658-2b4a2880-b8af-11eb-9a89-feb7e2342d6c.png)

BTW I've no idea if it came to be Ctrl+Shift+S in Notepad++ because of the way it is in VS, or some other reason.
And I don't much care.
The judgment is still a lack of an overwhelming standard to drive a change.
Thus, for legions of Notepad++ users that are used to the way things are, it won't be changed in how Notepad++ is shipped.

**Comment by sasumner** [comment]
>if I had presented 25 programs and only 24 would have used the same shortcut, your reply would have been the same.

No, a ratio like that would be a huge amount of evidence supporting a "standard".

**Comment by mere-human** [comment]
@kriegaex Remember, there are still a lot of users that are used to the current shortcut.
Besides, it can be changed in preferences:
![image](https://user-images.githubusercontent.com/9664141/118871764-8cdeb700-b8f0-11eb-858d-43f0087bdd23.png)

**Comment by kriegaex** [comment]
> > In none of those applications Ctrl-Shift-S was "Save all".
> 
> Arguably the most widely used programmer's text editor in the world is Visual Studio's -- which for those that don't know is NOT "Visual Studio Code" -- and it brings to the table:
> 
> ![image](https://user-images.githubusercontent.com/30118311/118865658-2b4a2880-b8af-11eb-9a89-feb7e2342d6c.png)

I don't know where you got that from. Maybe you customised it that way on your own workstation because you wanted it to be like in Notepad++, but the default is Strg-Shift-S for "Save All", like I said. Ctrl-K S is "Save all". This is a screenshot from my uncustomised VS Code:

![image](https://user-images.githubusercontent.com/1537384/118901276-10be9080-b93d-11eb-9757-85789c496061.png)

And here is the [official keyboard shortcut overview](https://code.visualstudio.com/shortcuts/keyboard-shortcuts-windows.pdf).

**Comment by kriegaex** [comment]
> @kriegaex Remember, there are still a lot of users that are used to the current shortcut.

True, for them it would be a breaking change. But they would get used to it and not have to remember this special shortcut, which is different from the two quasi standards found in the rest of the world.

> Besides, it can be changed in preferences:

Well, thanks, but did you read my original question? I already said there:

>> I stepped into that trap many times before until I finally decided to remap the keyboard to `F12` for "save as" (like in MS Office) and `Ctrl-Alt-S` for "save all". But that is just my personal solution, (...)

**Comment by GitMensch** [comment]
For me those issues are the same, but I want to also add one more:

When pressing the key-combo I was asked "do you want to save all", then I pressed ESC - and get still a save popup for every new file, which sounds like an easy to solve bug.
